homeassistant/components/azure_devops/sensor.py

```diff
@@ -1,126 +1,97 @@
-
-    for build in builds:
-        sensors.append(
-            AzureDevOpsLatestBuildSensor(client, organization, project, build)
+    coordinator, project = hass.data[DOMAIN][entry.entry_id]
+
+    sensors = [
+        AzureDevOpsSensor(
+            coordinator,
+            AzureDevOpsSensorEntityDescription(
+                key=f"{build.project.id}_{build.definition.id}_latest_build",
+                name=f"{build.project.name} {build.definition.name} Latest Build",
+                icon="mdi:pipe",
+                attrs=lambda build: {
+                    "definition_id": build.definition.id,
+                    "definition_name": build.definition.name,
+                    "id": build.id,
+                    "reason": build.reason,
+                    "result": build.result,
+                    "source_branch": build.source_branch,
+                    "source_version": build.source_version,
+                    "status": build.status,
+                    "url": build.links.web,
+                    "queue_time": build.queue_time,
+                    "start_time": build.start_time,
+                    "finish_time": build.finish_time,
+                },
+                build_key=key,
+                organization=entry.data[CONF_ORG],
+                project=project,
+                value=lambda build: build.build_number,
+            ),
         )
+        for key, build in enumerate(coordinator.data)
+    ]
 
     async_add_entities(sensors, True)
 
 
 class AzureDevOpsSensor(AzureDevOpsDeviceEntity, SensorEntity):
-    """Defines a Azure DevOps sensor."""
+    """Define a Azure DevOps sensor."""
 
     def __init__(
         self,
-        client: DevOpsClient,
-        organization: str,
-        project: str,
-        key: str,
-        name: str,
-        icon: str,
-        measurement: str = "",
-        unit_of_measurement: str = "",
+        coordinator: DataUpdateCoordinator,
+        description: AzureDevOpsSensorEntityDescription,
     ) -> None:
         """Initialize Azure DevOps sensor."""
-        self._attr_native_unit_of_measurement = unit_of_measurement
-        self.client = client
-        self.organization = organization
-        self.project = project
-        self._attr_unique_id = "_".join([organization, key])
-
-        super().__init__(organization, project, name, icon)
-
-
-class AzureDevOpsLatestBuildSensor(AzureDevOpsSensor):
-    """Defines a Azure DevOps card count sensor."""
-
-    def __init__(
-        self, client: DevOpsClient, organization: str, project: str, build: DevOpsBuild
-    ) -> None:
-        """Initialize Azure DevOps sensor."""
-        self.build: DevOpsBuild = build
         super().__init__(
-            client,
-            organization,
-            project,
-            f"{build.project.id}_{build.definition.id}_latest_build",
-            f"{build.project.name} {build.definition.name} Latest Build",
-            "mdi:pipe",
+            coordinator,
```

Comment:
now you can remove this init

Reply:
I get this lint error if I do this?

```
homeassistant/components/azure_devops/sensor.py:72: error: Definition of "entity_description" in base class "AzureDevOpsEntity" is incompatible with definition in base class "SensorEntity"  [misc]
homeassistant/components/azure_devops/sensor.py:78: error: "AzureDevOpsEntityDescription" has no attribute "build_key"  [attr-defined]
homeassistant/components/azure_devops/sensor.py:79: error: "AzureDevOpsEntityDescription" has no attribute "value"  [attr-defined]
homeassistant/components/azure_devops/sensor.py:84: error: "AzureDevOpsEntityDescription" has no attribute "build_key"  [attr-defined]
homeassistant/components/azure_devops/sensor.py:85: error: "AzureDevOpsEntityDescription" has no attribute "attrs"  [attr-defined]
```

Reply:
Handled in https://github.com/home-assistant/core/pull/54978/commits/bcc36fd452c1114bc5a166906735dcd886de7a7a